src/features/tokens/SelectTokenIdField.tsx

```diff
@@ -0,0 +1,108 @@
+import { useField } from 'formik';
+import Image from 'next/image';
+import { useState } from 'react';
+
+import ChevronIcon from '../../images/icons/chevron-down.svg';
+import { Modal } from '../../components/layout/Modal';
+import { useTokenIdBalance } from './useTokenBalance';
+import { useAccount } from 'wagmi';
+import { Spinner } from '../../components/animation/Spinner';
+
+type Props = {
+  name: string;
+  chainId: ChainId;
+  tokenAddress: Address;
+  disabled?: boolean;
+};
+
+export function SelectTokenIdField({ name, chainId, tokenAddress, disabled }: Props) {
+  const [, , helpers] = useField<number>(name);
+  const [tokenId, setTokenId] = useState<string | undefined>(undefined)
+  const { address } = useAccount();
+  const handleChange = (newTokenId: string) => {
+    helpers.setValue(parseInt(newTokenId));
+    setTokenId(newTokenId)
+  };
+
+  const { isLoading, tokenIds } = useTokenIdBalance(chainId, tokenAddress, address)
+
+  const [isModalOpen, setIsModalOpen] = useState(false);
+
+  const onClick = () => {
+    if (!disabled) setIsModalOpen(true);
+  };
+
+  return (
+    <div className="flex flex-col items-center">
```

Comment:
This component must show an input text field if the token contract does not support `tokenOfOwnerByIndex`